### PR TITLE
Bump app services dep, and include logins API changes

### LINKED ISSUE #1690: Bump A-S to 0.13.1
A-S just cut a new release which fixes a few issues that we notice in `reference-browser`. Let's bump our dependency to fix these in `r-b`, and look out for any regressions.

### PR DESCRIPTION
**Pull request by thomcc**
@csadilek, @grigoryk 

Fixes #1690

### CODE REVIEW

**Comment by grigoryk** [comment]
0.13.0 breaks FxA (manifests as IO exceptions when talking to network), @thomcc is investigating.

File: components/service/sync-logins/README.md [hunk 1]
```diff
@@ -161,6 +161,12 @@ Currently there is no way to change the key, once set (see https://github.com/mo
 
 Part is in this tree, but most is in https://github.com/mozilla/application-services. Specifically, there are two pieces not present in this repository, an [android-specific piece written in Kotlin](https://github.com/mozilla/application-services/tree/master/logins-sql/tree/master/logins-api/android), and a [cross-platform piece written in Rust](https://github.com/mozilla/application-services/tree/master/logins-sql).
 
+### What's the difference between `reset`, `wipe` and `wipeLocal`?
+
+- `reset` deletes last sync timestamps, and what the database believes is stored remotely. This is almost never what you want, and so it is deprecated.
+- `wipe` deletes all records, but remembers that it has deleted them. The next time we sync, we'll delete the remote versions of the deleted records too (e.g. if you wipe, then another client uploads a record, and then you sync, the other clients record will not be deleted).
+- `wipeLocal` deletes all local data from the database, bringing us back to the state prior to the first sync.
```
Comment (grigoryk) [inline components/service/sync-logins/README.md]: ... and prior to first local write, as well? E.g. it'll be just an empty database? As written, one might thing that only "from sync" data will be wiped.

File: components/service/sync-logins/README.md [hunk 2]
```diff
@@ -161,6 +161,12 @@ Currently there is no way to change the key, once set (see https://github.com/mo
 
 Part is in this tree, but most is in https://github.com/mozilla/application-services. Specifically, there are two pieces not present in this repository, an [android-specific piece written in Kotlin](https://github.com/mozilla/application-services/tree/master/logins-sql/tree/master/logins-api/android), and a [cross-platform piece written in Rust](https://github.com/mozilla/application-services/tree/master/logins-sql).
 
+### What's the difference between `reset`, `wipe` and `wipeLocal`?
```
Comment (grigoryk) [inline components/service/sync-logins/README.md]: Thanks for documenting these!

File: components/service/sync-logins/README.md [hunk 3]
```diff
@@ -161,6 +161,12 @@ Currently there is no way to change the key, once set (see https://github.com/mo
...
+
+- `reset` deletes last sync timestamps, and what the database believes is stored remotely. This is almost never what you want, and so it is deprecated.
```
Comment (grigoryk) [inline components/service/sync-logins/README.md]: Do we have any consumers even using it `reset`? I'd just remove it if not...

File: components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt [hunk 4]
```diff
@@ -111,9 +111,21 @@ interface AsyncLoginsStorage : AutoCloseable {
      *
      * @throws InvalidKeyException if the encryption key is wrong, or the db is corrupt
      * @throws MismatchedLockException if we're already unlocked
+     * @throws LoginsStorageException On unexpected errors (IO failure, rust panics, etc)
      */
```
Comment (grigoryk) [inline components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt]: Perhaps we actually want to use `@Throws` annotation?
> This annotation indicates what exceptions should be declared by a function when compiled to a JVM method.

File: components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt [hunk 5]
```diff
@@ -311,24 +391,13 @@ data class SyncableLoginsStore(
...
-        // Instead of synchronizing on 'store' let's be optimistic and just swallow mismatched locks.
 
-        if (store.isLocked()) {
-            try {
-                store.unlock(key().await()).await()
-            } catch (e: MismatchedLockException) {
-                // Oh well, it's already unlocked!
-            }
-        }
+        store.ensureUnlocked(key().await()).await()
```
Comment (grigoryk) [inline components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt]: At face value, this looks unsafe to me. Does anything prevent another thread from locking the store after you've ensured that it's unlocked and before you operate on it?

I'd prefer to see something like...
```
store.whileUnlocked(key().await()) {
    return block(store)
}
```
... and let the underlying implementation deal with locking, mutexing, whatever.

File: components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt (hunk 4 above)
Comment (thomcc) [reply components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt]: I'll rename it to @rejectsWith, or something.

File: components/service/sync-logins/README.md (hunk 3 above)
Comment (thomcc) [reply components/service/sync-logins/README.md]: Hm, well, yes, lockbox uses it. But they should use `wipeLocal` instead. I don't mind taking it out here.

File: components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt (hunk 5 above)
Comment (thomcc) [reply components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt]: After discussion in slack, this is tricky and should be done in a follow up if desired.

File: components/service/sync-logins/README.md (hunk 2 above)
Comment (thomcc) [reply components/service/sync-logins/README.md]: Okay, I'm keeping the mention of `reset` to make it clear that it's not the case that reset was renamed to wipeLocal (while being very clear that it no longer exists).

**Pull request merged**